src/k2/cpo/client/CPOClient.h

```diff
@@ -77,17 +77,18 @@ class CPOClient {
     // is not assigned or if there was a retryable error. It allows only one outstanding
     // request for a given collection.
     template <typename ClockT=Clock>
-    seastar::future<Status> GetAssignedPartitionWithRetry(Deadline<ClockT> deadline, const String& name, const dto::Key& key, uint8_t retries = 1) {
+    seastar::future<Status> GetAssignedPartitionWithRetry(Deadline<ClockT> deadline, const String& name, const dto::Key& key, 
+                                    bool reverse = false, bool excludedKey = false, uint8_t retries = 1) {
```

Comment:
Minor: parameter naming conventions - 
1. Try to avoid generic variable name unless the scope/context is well defined - "name" will be much better to be collectionName as this function param here.  While in CollectionGetRequest, data member "name" is ok, since its context is clearly for a collection, and if choosing "collectionName" here will even be more redundant. 
2. In general, Boolean param should start with sth like "is...", "has...", "contains..." etc, unless it is subjective and obvious, here, "reverse" may be ok, excludedKey for sure is not clear as "isExcludedKey". 

Also, as "reverse" only make sense to be set to true when "isExcludedKey" is set to true. You may want to put excludedKey before reverse in the parameter list, and check when "reverse" is set, isExcludedKey must be true as well?